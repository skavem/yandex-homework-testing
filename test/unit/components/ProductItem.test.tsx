import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";
import React from "react";
import { ProductItem } from "../../../src/client/components/ProductItem";
import __Providers from "../__Providers";
import { basename, getPath } from "../helpers";

describe("ProductItem", () => {
  it("отображает товары, список которых приходит с сервера, когда товар 1", async () => {
    const app = (
      <__Providers basename={basename}>
        <ProductItem product={{ id: 1, name: "Товар 1", price: 100 }} />
      </__Providers>
    );
    const { container, getAllByTestId } = render(app);

    const item = getAllByTestId("1")[0];

    await waitFor(() => expect(item).toBeInTheDocument());
    expect(
      item.getElementsByClassName("card-title")[0].innerHTML.includes("Товар 1")
    ).toBe(true);
    expect(
      item.getElementsByClassName("card-text")[0].innerHTML.includes("100")
    ).toBe(true);
    expect(item.getElementsByTagName("a")[0].getAttribute("href")).toBe(
      getPath("/catalog/1", false)
    );
  });
});
