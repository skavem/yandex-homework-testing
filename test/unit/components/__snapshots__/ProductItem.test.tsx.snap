// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem отображает товары, список которых приходит с сервера, когда товар 1 1`] = `
<div>
  <div
    class="ProductItem card w-100 mb-4"
    data-testid="1"
  >
    <img
      class="Image card-img-top"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
    <div
      class="card-body"
    >
      <h5
        class="ProductItem-Name card-title"
      >
        Товар 1
      </h5>
      <p
        class="ProductItem-Price card-text"
      >
        $
        100
      </p>
      <a
        class="ProductItem-DetailsLink card-link"
        href="/hw/store/catalog/1"
      >
        Details
      </a>
    </div>
  </div>
</div>
`;
